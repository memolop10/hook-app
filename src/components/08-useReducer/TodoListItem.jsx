import React from 'react'

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {

    return (
        
            <li 
                className="list-group-item" 
                key={ todo.id }
            >
                <p className={ `${ todo.done && 'complete' }` } 
                    onClick={ () => handleToggle( todo.id ) }
                > 
                        { index + 1 }. { todo.desc } 
                </p>
                <button 
                    className="btn btn-danger"
                    onClick={ () => handleDelete( todo.id ) } 
                >
                    Delete
                </button>
            </li>
    )
}

export default TodoListItem
