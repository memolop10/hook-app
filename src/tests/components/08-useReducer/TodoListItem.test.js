import { shallow } from "enzyme"
import TodoListItem from "../../../components/08-useReducer/TodoListItem"
import { demoTodos } from "../fixtures/demoTodos"



describe('Test in <TodoListItem />', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn()

    const wrapper = shallow( <TodoListItem 
        todo={ demoTodos[0] }
        index={ 11 }
        handleDelete= { handleDelete }
        handleToggle= { handleToggle }
    /> )

    test('should render correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('should call function handleDelete', () => {
        
        wrapper.find('button').simulate('click');
        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );

    })
    
    test('should call function handleToggle', () => {
        
        wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );
        
    })

    test('should show the texto correctly', () => {
        
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe( `12. ${ demoTodos[0].todo }`);
        
    })

    test('should have class complete if the TODO.done = true ', () => {
        
        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow( <TodoListItem 
            todo={ todo }
        /> )

        expect( wrapper.find('p').hasClass('complete') ).toBe(true);
    
    })
    
    
})
