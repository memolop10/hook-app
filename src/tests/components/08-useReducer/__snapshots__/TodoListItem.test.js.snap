// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoListItem /> should render correctly 1`] = `
<li
  className="list-group-item"
  key="11"
>
  <p
    className="false"
    onClick={[Function]}
  >
    12
    . 
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
